src/Integration.Vsix/Helpers/IStatusBarNotifier.cs

```diff
@@ -35,20 +36,24 @@ internal interface IStatusBarNotifier
     [PartCreationPolicy(CreationPolicy.Shared)]
     internal class StatusBarNotifier : IStatusBarNotifier
     {
-        private IVsStatusbar vsStatusBar;
+        private readonly IVsUIServiceOperation vSServiceOperation;
+        private readonly IThreadHandling threadHandling;
 
         [ImportingConstructor]
-        public StatusBarNotifier([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
+        public StatusBarNotifier(IVsUIServiceOperation vSServiceOperation, IThreadHandling threadHandling)
```

Comment:
I know in general we seperate stuff but to me it made sense to add the IThreadHandling in here and replace the ThreadHandling.Run